### DISCUSSION TITLE
[Redux/React] Qual a recomendação de quando usar o Redux Saga e o Reselect?

### DISCUSSION THREAD

**Discussion by MatheusNascimentoAraujo** [frontendbr/forum#1192] [Perguntas e Respostas]

**Comment by joaolavoier-luizalabs** [comment]
Olá Matheus. 

De uma maneira bem simples e resumida. São bibliotecas funcionais. Vou deixar em negrito algumas palavras chaves, caso não tenha conhecimento para você pesquisar. 

**Redux-Saga: ** Biblioteca para organizar os efeitos colaterais(**side-effects**) de sua aplicação. 
> O Redux é uma biblioteca funcional(**functional programming principles**), ou seja, as actions devem ser puras(**pure functions**). Porém você uma hora vai precisa fazer uma chamada assíncrona. Sem o redux-saga era muito comum criar **async actions** para realizar essas chamadas, isso deixa a aplicação bem mais difícil de debugar e até de testar estas actions. O Redux-Saga organiza os side-effects de maneira reativa, é um middleware do redux que fica escutando suas actions, e te permite fazer chamadas assíncronas e também de outras actions de sua aplicação sem tirar a pureza de suas actions. Também é uma abordagem mais fácil de ser testada. Exige um pouco de conhecimento sobre **javascript generators**.

**Reselect:** Biblioteca para melhorar a performance da aplicação nas alterações de estado
> Nunca utilizei esta biblioteca, porém pelo que sei dela é que ela cria um **memoize** dos estados para não precisar alterar os estados de um componente sem a necessidade. 

Espero ter ajudado.
